akka-actor/src/main/resources/reference.conf

```diff
@@ -351,16 +351,61 @@ akka {
       }
 
       affinity-pool-executor {
-        # The number of threads that this executor will maintain in its pool
-        num-threads = 8
+        # Min number of threads to cap factor-based parallelism number to
+        parallelism-min = 8
+
+        # The parallelism factor is used to determine thread pool size using the
+        # following formula: ceil(available processors * factor). Resulting size
+        # is then bounded by the parallelism-min and parallelism-max values.
+        parallelism-factor = 3.0
+
+        # Max number of threads to cap factor-based parallelism number to
+        parallelism-max = 64
 
-        # The max size of the queue that is used by each thread in the pool
-        # If the queue of a particular worker is full, no more tasks can be
-        # enqueued until space frees up
+        # Each worker in the pool uses a seprate bounded MPSC queue. This value
+        # indicates the upper bound of the queue. Whenever an attempt to enqueue
+        # a task is made and the queue does not have capacity to accomodate
+        # the task, RejectedExecutionException is thrown.
```

Comment:
Might be worth making the `on-overflow` strategy a configuration parameter, with the initally only available strategy being `throw-rejected-execution`?